stacks-signer/src/monitoring/mod.rs

```diff
@@ -0,0 +1,174 @@
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::COORDINATOR_INBOUND_MESSAGES.inc_by(amount);
+}
+
+/// Increment the number of inbound packets received
+#[allow(unused_variables)]
+pub fn increment_inbound_packets(amount: i64) {
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::INBOUND_PACKETS_RECEIVED.inc_by(amount);
+}
+
+/// Increment the number of commands processed
+#[allow(unused_variables)]
+pub fn increment_commands_processed(command_type: &str) {
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::COMMANDS_PROCESSED
+        .with_label_values(&[command_type])
+        .inc();
+}
+
+/// Increment the number of DKG votes submitted
+#[allow(unused_variables)]
+pub fn increment_dkg_votes_submitted() {
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::DGK_VOTES_SUBMITTED.inc();
+}
+
+/// Increment the number of commands processed
+#[allow(unused_variables)]
+pub fn increment_operation_results(operation_type: &str) {
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::OPERATION_RESULTS
+        .with_label_values(&[operation_type])
+        .inc();
+}
+
+/// Increment the number of block proposals received
+#[allow(unused_variables)]
+pub fn increment_block_proposals_received() {
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::BLOCK_PROPOSALS_RECEIVED.inc();
+}
+
+/// Update the stx balance of the signer
+#[allow(unused_variables)]
+pub fn update_signer_stx_balance(balance: i64) {
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::SIGNER_STX_BALANCE.set(balance);
+}
+
+/// Update the signer nonce metric
+#[allow(unused_variables)]
+pub fn update_signer_nonce(nonce: u64) {
+    #[cfg(feature = "monitoring_prom")]
+    prometheus::SIGNER_NONCE.set(nonce as i64);
+}
+
+/// Start a new RPC call timer.
+/// The `origin` parameter is the base path of the RPC call, e.g. `http://node.com`.
+/// The `origin` parameter is removed from `full_path` when storing in prometheus.
+#[cfg(feature = "monitoring_prom")]
+pub fn new_rpc_call_timer(full_path: &str, origin: &String) -> HistogramTimer {
+    let path = &full_path[origin.len()..];
+    let histogram = prometheus::SIGNER_RPC_CALL_LATENCIES_HISTOGRAM.with_label_values(&[path]);
+    histogram.start_timer()
+}
+
+/// Start serving monitoring metrics.
+/// This will only serve the metrics if the `monitoring_prom` feature is enabled.
+#[allow(unused_variables)]
+pub fn start_serving_monitoring_metrics(config: GlobalConfig) -> Result<(), String> {
+    #[cfg(feature = "monitoring_prom")]
+    {
+        if config.metrics_endpoint.is_none() {
+            return Ok(());
+        }
+        let thread = std::thread::Builder::new()
+            .name("signer_metrics".to_string())
+            .spawn(move || {
+                if let Err(monitoring_err) = server::MonitoringServer::start(&config) {
+                    error!(
+                        "Monitoring: Error starting metrics server: {:?}",
```

Comment:
This error isn't just during `starting`, because `start()` also invokes the server mainloop, `Error in metrics server: ...` may be a better message.